File: framework/languages/java/e2e-kotlin/AutomaticTurnDispatch/Server/Session/src/main/java/systems/zlink/e2e/kotlin/automaticturn/SessionApplication.java
package systems.zlink.e2e.kotlin.automaticturn;

import org.springframework.boot.WebApplicationType;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.annotation.Bean;
import systems.zlink.contracts.core.RoutingId;
import systems.zlink.framework.configuration.ZLinkMessageFlowLogMode;
import systems.zlink.framework.configuration.ZLinkSpotNodeBuilder;
import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions;
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore;
import systems.zlink.framework.spring.EnableZLinkFramework;
import systems.zlink.framework.spring.ZLinkFrameworkConfigurer;

@EnableZLinkFramework
@SpringBootApplication(proxyBeanMethods = false)
public final class SessionApplication {
    private SessionApplication() {
    }

    public static AutoCloseable run(String... args) {
        SpringApplicationBuilder builder = new SpringApplicationBuilder(SessionApplication.class)
            .web(WebApplicationType.NONE);
        builder.application().setKeepAlive(true);
        return builder.run(args)::close;
    }

    @Bean
    ZLinkFrameworkConfigurer framework() {
        return options -> {
            String nodeRid = Env.get("nodeRid", "session-a");
            String logDir = Env.get("logDirectory", "logs");
            options.configureDispatch()
                .messageFlow(ZLinkMessageFlowLogMode.KEY_TRANSITIONS)
                .traceLogFile(logDir + "/session-flow.log")
                .traceLabel("kotlin-atd-session");
            var route = options.addRouteMeshChannel(Contracts.SPOT_MESH)
                .enableServer(Env.get("sessionRouteEndpoint"))
                .enableClient(Env.get("playRouteEndpoint"))
                .setRoutingId(RoutingId.from(nodeRid));
            String playBRoute = Env.get("playBRouteEndpoint", "");
            if (!playBRoute.isBlank()) {
                route.enableClient(playBRoute);
            }
            ZLinkSpotNodeBuilder spot = options.addSpotMesh(Contracts.SPOT_MESH)
                .enableRouter(Env.get("sessionSpotEndpoint"))
                .setRoutingId(RoutingId.from(nodeRid));
            spot.addEntrySpot(ProbeEntrySpot.class);
            spot.addActorFactory("probe", ProbeActorFactory.class);
            options.addStreamNode("gateway")
                .bind(Env.get("streamEndpoint"))
                .registerSession(ProbeSession.class)
                .addSessionPacketHandler(ActorAuthReqHandler.class)
                .addSessionPacketHandler(BindActorsReqHandler.class)
                .addSessionPacketHandler(EnsureSpotReqHandler.class)
                .addSessionPacketHandler(RemoteSpotAwaitReqRouteHandler.class)
                .addSessionPacketHandler(ShutdownAwaitReqRouteHandler.class)
                .addSessionPacketHandler(ShutdownRecoveryReqRouteHandler.class)
                .addSessionPacketHandler(ProbeReqRouteHandler.class)
                .addSessionPacketHandler(CleanupProbeReqRouteHandler.class)
                .addSessionPacketHandler(HoldMsgRouteHandler.class)
                .addSessionPacketHandler(AwaitMsgRouteHandler.class)
                .addSessionPacketHandler(WorkerAwaitMsgRouteHandler.class)
                .addSessionPacketHandler(ProbeMsgRouteHandler.class)
                .addSessionPacketHandler(TimerStartMsgRouteHandler.class)
                .addSessionPacketHandler(TimerStopMsgRouteHandler.class)
                .addSessionPacketHandler(AwaitTimeoutMsgRouteHandler.class)
                .addSessionPacketHandler(AwaitTimeoutReqRouteHandler.class)
                .addSessionPacketHandler(AwaitCancelMsgRouteHandler.class)
                .addSessionPacketHandler(SpotProbeMsgRouteHandler.class)
                .addSessionPacketHandler(EvidenceReqRouteHandler.class);
        };
    }

    @Bean
    ZLinkRedisLocationStore locationStore() {
        return new ZLinkRedisLocationStore(new ZLinkRedisLocationOptions()
            .setConnectionString(Env.get("redisLocationEndpoint"))
            .setKeyPrefix(Env.get("locationKeyPrefix")));
    }
}

File: framework/languages/java/samples/kotlin/SupportChat/Server/Session/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/session/SessionApplication.kt
package systems.zlink.samples.kotlin.supportchat.server.session

import kotlinx.coroutines.Dispatchers
import org.springframework.boot.WebApplicationType
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.annotation.Bean
import systems.zlink.framework.configuration.ZLinkMessageFlowLogMode
import systems.zlink.framework.kotlin.configureDispatch
import systems.zlink.framework.kotlin.useCoroutineHandlers
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore
import systems.zlink.framework.spring.EnableZLinkFramework
import systems.zlink.framework.spring.ZLinkFrameworkConfigurer
import systems.zlink.samples.kotlin.supportchat.server.configuration.SampleFlowLog
import systems.zlink.samples.kotlin.supportchat.server.configuration.SampleLocationStore
import systems.zlink.samples.kotlin.supportchat.server.configuration.SampleNames
import systems.zlink.samples.kotlin.supportchat.server.configuration.SampleTopology
import systems.zlink.samples.kotlin.supportchat.server.session.sessions.SupportChatSession

@EnableZLinkFramework
@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackageClasses = [SessionApplication::class],
)
class SessionApplication {
    @Bean
    fun sessionFramework(): ZLinkFrameworkConfigurer =
        ZLinkFrameworkConfigurer { options ->
            val topology = SampleTopology.create()
            val session = topology.primarySession
            options.addHandlersFromPackageOf(SessionApplication::class.java)
            options.useCoroutineHandlers(Dispatchers.Default)
            options.configureDispatch {
                messageFlow(ZLinkMessageFlowLogMode.KEY_TRANSITIONS)
                traceLogFile(SampleFlowLog.path("session"))
                traceLabel("session")
            }
            options.addClientServerChannel(SampleNames.ApiChannel)
                .enableClient()
            options.addClientServerChannel(SampleNames.SupportChannel)
                .enableClient()
            val node = options.addSpotMesh(SampleNames.SupportSpotDiscovery)
            node.enableRouter(session.routerEndpoint)
                .setRoutingId(session.routingId)
            node.enablePubSub(session.pubEndpoint)
            options.addStreamNode(SampleNames.StreamNode)
                .bind(session.streamEndpoint)
                .registerSession(SupportChatSession::class.java)
        }

    @Bean
    fun locationStore(): ZLinkRedisLocationStore = SampleLocationStore.create()

    companion object {
        fun run(args: Array<String> = emptyArray()): AutoCloseable {
            val builder = SpringApplicationBuilder(SessionApplication::class.java)
                .web(WebApplicationType.NONE)
            builder.application().setKeepAlive(true)
            val context = builder.run(*args)
            return AutoCloseable { context.close() }
        }
    }
}
